cts/cts-regression.in

```diff
@@ -1,230 +1,335 @@
-                run_as_root $test_home/cts-exec -R
-                rc=$?
-            else
-                error "pacemaker_remote regression test not found"
-                rc=$CRM_EX_NOT_INSTALLED
-            fi
-            ;;
-        fencing)
-            if [ -x $test_home/cts-fencing ]; then
-                run_as_root $test_home/cts-fencing
-                rc=$?
-            else
-                error "fencing regression test not found"
-                rc=$CRM_EX_NOT_INSTALLED
-            fi
-            ;;
-        cli)
-            if [ -x $test_home/cts-cli ]; then
-                $test_home/cts-cli $verbose $valgrind
-                rc=$?
-            else
-                error "cli regression test not found"
-                rc=$CRM_EX_NOT_INSTALLED
-            fi
-            ;;
-    esac
-
-    info "============================================================"
-    info ""
-    info ""
-    return $rc
-}
-
-run_tests() {
-    local TEST
-    local TEST_RC
-    local FAILED
-
-    FAILED=""
-    for TEST in "$@"; do
-        run_test $TEST
-	TEST_RC=$?
-        if [ $TEST_RC -ne 0 ]; then
-            info "$TEST regression tests failed ($TEST_RC)"
-            FAILED="$FAILED $TEST"
-        fi
-    done
-    if [ -n "$FAILED" ]; then
-        error "failed regression tests: $FAILED"
-        return $CRM_EX_ERROR
-    fi
-    return $CRM_EX_OK
-}
-
-while [ $# -gt 0 ] ; do
-    case "$1" in
-        --help)
-            echo "$USAGE_TEXT"
-            exit $CRM_EX_OK
-            ;;
-        -V|--verbose)
-            verbose="-V"
-            shift
-            ;;
-        -v|--valgrind)
-            valgrind="-v"
-            shift
-            ;;
-        scheduler|exec|fencing|cli)
-            add_test $1
-            shift
-            ;;
-        pacemaker_remote)
-            if [ $REMOTE_ENABLED -eq 0 ]; then
-                error "Pacemaker Remote not supported by this build"
-                exit $CRM_EX_USAGE
-            else
-                add_test $1
-                shift
-            fi
-            ;;
-        all)
-            add_test scheduler
-            add_test cli
-            add_test exec
-            add_test fencing
-            shift
-            ;;
-        *)
-            error "unknown option: $1"
-            echo
-            echo "$USAGE_TEXT"
-            exit $CRM_EX_USAGE
-            ;;
-    esac
-done
-
-if [ -z "$tests" ]; then
-    add_test scheduler
-    add_test cli
-fi
-
-run_tests $tests
+#!@PYTHON@
+"""Convenience wrapper for running Pacemaker regression tests.
+
+Usage: cts-regression [-h] [-V] [-v] [COMPONENT ...]
+"""
+
+__copyright__ = 'Copyright 2012-2022 the Pacemaker project contributors'
+__license__ = 'GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY'
+
+import argparse
+from enum import IntEnum, unique
+import os
+import subprocess
+import sys
+import textwrap
+
+
+REMOTE_ENABLED = bool('@PC_NAME_GNUTLS@')
+TEST_HOME = ''
```

Comment:
Not sure this should be a "constant" (all caps, no true constants) in hindsight, since we set it in `main()`. Doesn't even have to be global. We could pass it as a constructor argument to `Component`. Feels redundant that way but perhaps it's more semantically correct.